common/web/types/src/kmx/kmx-plus/kmx-plus.ts

```diff
@@ -161,6 +173,8 @@ export interface StrsOptions {
   nfd?: boolean;
   /** string can be stored as a single CharStrsItem, not in strs table. */
   singleOk?: boolean;
+  /** optional context */
+  x?: any;
```

Comment:
Why named `x`? Also 'context' is a bit of an overloaded term, so I wonder if we can give a longer name, such as `compileContext`?

Reply:
It's `x` like the other parameter to `m()` is traditionally named `o`

Perhaps `ObjectWithMetadata` should be renamed `ObjectWithCompileContext` ?

Reply:
> Perhaps `ObjectWithMetadata` should be renamed `ObjectWithCompileContext` ?

Yes, that sounds good to me.

> It's `x` like the other parameter to `m()` is traditionally named `o`

See https://github.com/keymanapp/keyman/issues/14117#issuecomment-2942248149